bip-taro.mediawiki

```diff
@@ -470,6 +589,67 @@ other normal script-path scripts within the tapscript tree.
 The above example creates only a single asset. It's possible to also create
 several assets within a single output, thereby batching asset creation.
 
+===Asset Burning===
+
+Given the requirement that Taro commitments be unique within the context of a
+tapscript commitment, Taro assets can also be ''provably burned''.
+
+In order to burn an asset, the holder of the asset can simply ''remove'' the
+Taro asset leaf from their asset tree.
+
+In order to prove to a 3rd party that an asset has been burnt, the holder of
+said asset must reveal the following information:
+* A non-inclusion proof rooted at their asset tree that proves that asset in question is no longer committed to.
+* A proof for each other top-level output in the burning transaction that asserts the following:
+** For each output, the first depth of the control block leaf reveal must be presented:
+*** If the output key was created via BIP 86, then simply verify the mapping based on the internal key. In this case there is no script tree being committed to.
+*** If the proof is of depth 0, meaning only a single value is committed to, then assert the item isn't a valid Taro asset root.
+*** If the proof is of depth 1, then assert that either both elements are tap branches, or one of the items is a leaf that isn't a valid Taro asset root.
```

Comment:
Here we disallow ANY other Taro asset roots in the TX vs. above; IIUC this implies that an asset burn must be the only Taro-related activity in a given TX? (Cannot issue new assets owned by another output, or perform a transfer to another output, etc.).

Reply:
> IIUC this implies that an asset burn must be the only Taro-related activity in a given TX? (Cannot issue new assets owned by another output, or perform a transfer to another output, etc.).

Great question! The burn operation is allowed when batched with any other taro events (transfer, etc). The specific event here is defined by removal of a commitment from the sub-asset tree and therefore the main taro commitment. The burn verification routine has access to both the non-inclusion proof (verify it's no longer committed to in the simple case of a collectible), and also the _new_ taro root (to verify that this is placed in a unique location).

Reply:
These two additional rules are more or less summed up in the `taro_commitment_is_valid` routine, but I wanted to spell them out here as well. 

I made another updated to make this more explicit in that right now it sort of only allows a total burn (no assets committed to at all), vs a selective burn (some assets not committed to, or the committed amounts should be reduced).

Reply:
> Is it correct to equate:

Correct.